src/ros/mb_publisher.cpp

```diff
@@ -18,6 +18,36 @@ void MBPublisher::publish(ur_msgs::IOStates& io_msg, SharedMasterBoardData& data
   io_pub_.publish(io_msg);
 }
 
+void MBPublisher::publishRobotStatus(SharedRobotModeData& data)
+{
+  industrial_msgs::RobotStatus msg;
+  msg.drives_powered.val = data.robot_power_on;
```

Comment:
Can `robot_power_on` ever be `true` if `real_robot_enabled` is `false`? Perhaps in `ur-sim`?

Reply:
testing with ursim I get the following output in Poly:
0000d05h41m35.184s RobotInterface C102A0: Real Robot not connected - Simulating Robot

meanwhile the robot mode contains the following data:

when I turn off the robot:
robot_power_on: 0 physical robot: 0 real_robot: 1 mode: 3 (POWER_OFF)

when I turn on the robot but do not yet release the brakes (ON but not START):
robot_power_on: 1 physical robot: 0 real_robot: 1 mode: 5 (IDLE)

when I release the brakes:
robot_power_on: 1 physical robot: 0 real_robot: 1 mode: 7 (RUNNING)

so I'm not sure what real_robot_enabled is supposed to tell us, maybe there is a way to switch to moving a simulated robot on the actual controller?

Reply:
> so I'm not sure what `real_robot_enabled` is supposed to tell us, maybe there is a way to switch to moving a simulated robot on the actual controller?

There is a two-option selection bullet list at the bottom right or left (don't remember) that contains those two options. I've never used it, but IIRC you can use it to switch to a virtual robot.

Reply:
The difference between `IDLE` and `RUNNING` may indicate that `RUNNING` would be better indicator of `drives_powered`?

Reply:
ok, found that (for the reference: it is in 'Program Robot' -> 'Move' or 'I/O' but I can only change it to simulated in IO), real_robot_enabled then does turn to false. But this doesn't map to any of the industrial_msgs::RobotStatus fields does it?

Reply:
# Drive power status: True if drives are powered.  Motion commands will 
   # automatically enable the drives if required.  Drive power is not requred
   # for possible motion
   industrial_msgs/TriState drives_powered

So yes, maybe drives powered should be obtained from Mode == RUNNING as in that halfway on mode (breakes still on) it does not automatically start moving. But then drives_powered would be the same as motion_possible. And as far as I understand motion_possible is authorative?

Reply:
I think `drives_powered` should reflect the status of the drives (ie: `robot_power_on`.

`RUNNING` could then be used in the predicate for `motion_possible` (ie: `mode != RUNNING -> motion_possible = false`.